### ISSUE TITLE
False negative for `NullableToStringCall` rule in lambdas

### ISSUE THREAD

**Issue by jenssuhr** [detekt/detekt#6349]
## Expected Behavior

The `NullableToStringCall` rule should work the same inside lambda expressions (e.g. `let`) as it does everywhere else:
```kotlin
fun detected(): String {
    val x: String? = null
    return "hi $x"
}
```

Other rules like `ForbiddenMethodCall` also find noncompliant calls inside lambda expressions:
```kotlin
fun forbiddenDetected(): String = 1.let {
    "hi ${println("foo")}"
}
```

## Observed Behavior

Nullable variables in a string inside a lambda expressions are not detected by the rule.

## Steps to Reproduce

Call the `toString()` function (implicit or explicit) of a nullable value inside a lambda. Neither values defined inside nor outside the lambda trigger the rule.
```kotlin
fun undetected1(): String = 1.let {
    val x: String? = null
    "hi $x"
}

fun undetected2(): String = 1.let {
    val x: Any? = null
    "hi ${x.toString()}"
}

fun undetected3(): String {
    val x: String? = null
    return 1.let {
        "hi $x"
    }
}

fun undetected4(): String {
    val x: String? = null
    return x.let { y ->
        "hi $y"
    }
}
```

## Context

I was wondering why my custom rule that detects some forbidden calls inside string templates missed a noncompliant string in a lambda expression, and checked how the `NullableToStringCall` implementation handled it. As we were doing roughly the same things, I tested if the `NullableToStringCall` rule actually worked with lambda expressions.

## Your Environment

* Version of detekt used: 1.23.1
* Version of Gradle used (if applicable): 8.2.1
* Gradle scan link (add `--scan` option when running the gradle task): N/A
* Operating System and version: multiple (MacOS, Alpine Linux)
* Link to your project (if it's a public repository): N/A
